tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -51,6 +55,60 @@ bool GetEnvBool(const char* name, bool defval) {
 
 }  // namespace
 
+bool PrintFdMessage(long tgid) {
+  string path;
+  char line[100];
+  DIR* fd;
+  struct dirent* ent;
+  long deviceId;
+  path = absl::StrCat("/proc/", tgid, "/fd");
+  fd = opendir(path.c_str());
+  if (!fd) {
+    return false;
+  }
+  while (ent = readdir(fd)) {
+    if (!isdigit(*ent->d_name)) continue;
+    deviceId = strtol(ent->d_name, NULL, 10);
+    string paths;
+    int statusfd;
+    paths = absl::StrCat("/proc/", tgid, "/fd", deviceId);
+    statusfd = readlink(paths.c_str(), line, 100);
+    if (!statusfd) return false;
+    if (strncmp(line, "/dev/accel0", 11) != 0) continue;
+    closedir(fd);
```

Comment:
`fd` will leak if you take the `return false` above. Using the TF methods described below would fix this.

If that doesn't work, consider using std::unique_ptr with a custom deleter. See https://stackoverflow.com/a/19054467 for an example. I think you could do something like:
```c++
DIR* raw_fd_dir = opendir(path.c_str());
if (!raw_fd_dir) return false;
std::unique_ptr<DIR, int(*)(DIR*)> fd_dir(raw_fd_dir, closedir);
```

This is pretty fancy though, so if it doesn't work, another option is to pull out the logic that iterates through `fd` into a separate function, call that after opening it, and then call closedir(fd) before returning the result of the new function.